Kernel/FileSystem/File.h

```diff
@@ -124,7 +124,7 @@ class File
 
     void evaluate_block_conditions()
     {
-        if (Processor::current_in_irq()) {
+        if (Processor::current_in_irq() != 0) {
```

Comment:
What does the int mean that this function returns?

Reply:
It's used as a boolean in practice, but the m_in_irq field on Processor is kept as a FlatPtr for what I assume are padding/appointment purposes. Most of the fields are read using offsetof and the %gs segment register